pkg/kubelet/kubelet_pods.go

```diff
@@ -1143,23 +1143,27 @@ type PodKiller interface {
 	PerformPodKillingWork()
 	// After Close() is called, this pod killer wouldn't accept any more pod killing requests
 	Close()
-	// IsMirrorPodPendingTerminationByPodName checks whether the mirror pod for the given full pod name is pending termination
-	IsMirrorPodPendingTerminationByPodName(podFullname string) bool
-	// IsMirrorPodPendingTerminationByUID checks whether the mirror pod for the given uid is pending termination
-	IsMirrorPodPendingTerminationByUID(uid types.UID) bool
-	// MarkMirrorPodPendingTermination marks the mirror pod entering grace period of termination
-	MarkMirrorPodPendingTermination(pod *v1.Pod)
+	// IsPodPendingTerminationByPodName checks whether the mirror pod for the given full pod name is pending termination
+	IsPodPendingTerminationByPodName(podFullname string) bool
+	// IsPodPendingTerminationByUID checks whether the mirror pod for the given uid is pending termination
+	IsPodPendingTerminationByUID(uid types.UID) bool
+	// CheckAndMarkPodPendingTerminationByPod checks to see if a pod is being killed, if it isn't then add it
+	// It returns true if the pod is already being killed, false if it adds it to the queue
+	CheckAndMarkPodPendingTerminationByPod(*kubecontainer.PodPair) bool
```

Comment:
should this be part of the interface? seems like it is a helper that is only used internally.

Reply:
agree w/ @sjenning as this only appears to be called internally within the channel.

Reply:
Agreed. I moved out of the interface and made it private. fixed in the latest push.